## Problem 2

In [3]:
ls -lh #-lh is used to generate the long listing format with human readable size

total 2.4M
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  3 11:02  conv.txt
-rw-rw-r-- 1 osboxes osboxes  147 Sep  3 11:02  DISCLAIMER
-rw-rw-r-- 1 osboxes osboxes 1.8K Sep  5 09:26  Exam_1_Answers.ipynb
-rw-rw-r-- 1 osboxes osboxes 5.5K Sep  3 11:02 'Exam Week 1.ipynb'
-rw-rw-r-- 1 osboxes osboxes 3.9K Sep  3 11:02 'Exam Week 2.ipynb'
-rw-rw-r-- 1 osboxes osboxes 4.2K Sep  3 11:02  Germplasm.tsv
-rw-rw-r-- 1 osboxes osboxes  141 Sep  3 11:02  how_to_convert.txt
-rw-rw-r-- 1 osboxes osboxes 1.1K Sep  3 11:02  LICENSE
-rw-rw-r-- 1 osboxes osboxes  637 Sep  3 11:02  LocusGene.tsv
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  3 11:02  Locus_Germplasm_Phenotype_20130122.txt
-rw-rw-r-- 1 osboxes osboxes  743 Sep  3 11:02  README.md


## Ownership

The permissions of each file are shown within the first 10 characters of each entry, in this case -rw-rw-r--. Their meaning is explained in the following table:



  <center>Position </center> |  <center> Character </center>  |  <center> Meaning  </center> 
 --- | --- | --- 
  <center> 1 </center>  | <center> - </center> |   <center>Not a directory</center> 
  <center> 2 </center>  |  <center>r </center> |   <center>user has permission to read </center> 
  <center> 3  </center> |  <center>w </center> |   <center>user has permission to write </center> 
  <center> 4  </center> |  <center>- </center> |  <center> user does not have permission to execute </center> 
 <center>  5  </center> |  <center>r </center> |   <center>group has permission to read </center> 
  <center> 6 </center>  | <center> w </center> |   <center>group has permission to write </center> 
  <center> 7  </center> |  <center>- </center> |   <center>group does not have permission to execute </center> 
  <center> 8  </center> |  <center>r </center> |  <center> everybody has permission to read</center> 
  <center> 9  </center> |  <center>- </center> |   <center>not everybody has permission to write </center> 
  <center> 10  </center> |  <center>- </center> |   <center>not everybody has permission to execute</center> 

## Problem 3

In [6]:
head -1 Locus_Germplasm_Phenotype_20130122.txt

Locus_name)	Germplasm_name	phenotype	pubmed_id


The "head" command is used to show the first 10 lines of a file. 

"-1" means that I only want the first line to be shown. 

"Locus_Germplasm_Phenotype_20130122.txt" is the name of the file whose initial line will be shown.

## Problem 4

In [1]:
wc -l Locus_Germplasm_Phenotype_20130122.txt

7216 Locus_Germplasm_Phenotype_20130122.txt


"wc -l filename" is used to count the number of lines of a file.

## Problem 5

In [2]:
tail -n+2 Locus_Germplasm_Phenotype_20130122.txt > Data_only.csv

"tail" is used to show the last lines of a file.

"-n+2" indicates that every line after the second one (included) must be shown.

"Locus_Germplasm_Phenotype_20130122.txt" is the name of the file whose lines will be shown.

"> Data_only.csv" indicates that the output will be sent to the file Data_only.csv

In [13]:
wc -l Data_only.csv

7215 Data_only.csv


The expected number of lines is 7215, one fewer than the number of lines of the original file. 

The command "wc -l Data_only.csv" shows that there are 7215 lines.

## Problem 6

For this problem I have collaborated with Ismael Sánchez Pareja and Ana Mariya Anhel Valdes

In [10]:
grep -E -ia 'roots?,?\.?\s' Locus_Germplasm_Phenotype_20130122.txt

AT1G01550	bps1-2	Growth of bps1-2 mutants on CPTA-supplemented medium resulted in partial rescue of both leaf and root defects. bps1-2 mutants grown on control growth medium have small radialized leaves with very little vascular tissue, and very short misshapen knotted-looking roots. By contrast, bps1-2 mutants grown on CPTA-supplemented medium produced larger flattened leaves that contained primary and secondary veins, and smooth elongated roots.	17217459
AT1G01550	bps1-2	Mutant roots were abnormal; primary and lateral roots were short, root hairs formed close to the root apex, and root defects were most severe when plants were grown at low temperature.	15458645
AT1G01550	bps1-2	Under high light conditions (approximately 200 &mu;E.m-2.sec-1), CPTA-treated seedlings were completely photobleached and mutants showed rescue of both leaf and root development. Under moderate (approximately 100 &mu;E.m-2.sec-1) and low (30 &mu;E.m-2.sec-1) light intensities, seedlings are incompletely photob

AT1G08260	tilted1-4	Twenty-five percent of embryos show abnormal cell divisions during embryogenesis-large cells- decreased rate of cell division. Division of the hypohyseal cell abnormal leading to abnormal placement of the root pole. Homozygotes show delayed flowering, abnormal floraly phyllotaxy and abnormal ovules.	16278345
AT1G08370	dcp1-2	The homozygous progeny is seedling lethal, showed arrested postembryonic development including cotyledon expansion, development of vascular networks, root elongation, and shoot development.	17485080
AT1G08430	Atalm1-KO	On Al3+ medium, the mutant has significantly shorter roots than the Col wild type. In the absence of aluminum stress, the roots of mutant and wild type grew similarly.	16740662
AT1G08430	Atalm1-KO	The mutant exhibits very low levels of root  growth and malate release in the presence of  Al (0.2 nmol plant-1 24 h-1).
AT1G09100	rpt5b	Insensitive to high-glucose repression of cotyledon expansion, chlorophyll accumulation, true-leaf d

AT1G19220	CS24618	no obvious auxin-related growth phenotype, but roots show mild auxin resistance	15659631
AT1G19220	CS24625	The double mutant mutant displays agravitropic responses in both hypocotyls and roots. When seedlings are grown vertically under dark conditions, the regulation of growth orientation is disrupted in both hypocotyls and roots, with the hypocotyls occasionally growing downward and the roots upward. Also, the roots and hypocotyls of the double mutant show reduced gravitropic curvatures compared with the wild type when vertically dark-grown seedlings are reoriented by 90�.	15659631
AT1G19220	CS24625	The phenotype of the double mutant is most obvious at its seedling stage, with its most prominent phenotype being severely impaired  lateral root formation. Its primary roots fail to produce lateral roots in 2-week-old seedlings. However, double mutant seedlings start to generate several lateral roots after ~2 weeks of growth, and their morphological appearance is normal.

AT1G21970	CS3867	Embryo lethal; viviparous embryos with cotyledons partially transformed into leaves; embryos sensitive to abscisic acid; embryos have reduced hypocotyl, cotyledons that remain green late in development; reduced protein and lipid bodies with starch being predominant storage product in mutant embryos. Trichomes form on cotyledons after germination; internal cells of cotyledons enlarge, vacuolate and resemble leaf mesophyll cells; intercellular spaces are prominant in cotyledons; enlarged shoot apical meristem; stomata are found on surface of cotyledons of mutant embryos; mature seeds are desiccation intolerant, however, immature seeds can be rescued before silique drying and germinated to produce seedlings. Root apex is active in viviparous seeds; immature mutant embryos germinated prior to desiccation produce plants that appear normal for other characteristics.	12244233
AT1G21970	CS8101	Embryo lethal; viviparous embryos with cotyledons partially transformed into leaves;

AT1G29260	pex7-1	Displays peroxisomal defects such as reduced sensitivity to exogenously applied IBA.  Fewer lateral roots after IBA treatment.  Not generally defective in auxin responses.   Hypocotyl is not markedly shorter than WT when grown without added sucrose in the dark. PTS2-tageed proteins are not efficiently imported into the peroxisome.	15548601
AT1G29990	pdf6-1	Seedlings exhibit root swelling when grown in the presence of 100 nM oryzalin and have a 25% reduction in hypocotyl length. Reduced cell size is due to defects in the microtubules. Microtubules in the root elongation zone remain disorganized. Also affects epidermal cells which have abnormal division planes.	19004800
AT1G30270	cipk23-1	Enhanced drought tolerance. Hypersensitive to ABA inhibition  of light-induced stomatal opening. Loss of function of CIPK23 alters the ABA  responsiveness of guard cells during their opening and  closure, leading to reduced water loss under drought  conditions. When grown on media conta

AT1G49240	act2-1/act8-2	Lacks root hairs.Root hairs initiate but do not develop further.	19304937
AT1G49240	act8-2	Short root hairs. 50% reduction in root hair length compared to wild type.	19304937
AT1G49240	act8-2/act7-4	Plants are dwarf. Leaves are smaller and contain fewer cells and fewer lobes than wild type. Reduced number of trichome branches. Petals are narrow. Roots are more severely affected and produce short root hairs. Cell files are irregular and the elongation zone is much reduced.	19304937
AT1G50960	ga2ox7-2	Longer roots when plants are grown under salt stress (50-100nM NaCl).	18643985
AT1G51190	plt3-1/plt2-2/plt1-4	Rootless.Embryonic root differentiates 3 days after germination. Adventitious roots arrest 6 days after germination.	17960244
AT1G52340	aba2-2	Mutant showed a similar sensitivity to 25 mM trehalose as wild type (inhibition of root growth).	17031512
AT1G53140	SALK_065118	Grow normally when seeds were germinated under conventional conditions (21�C), mutant seed

AT1G58360	aap1-1	no obvious phenotype, endurance to toxic concentration of phenylalanine (2 mM) which did not influence development of roots and leaves of the mutant nor did a high level of phenylalanine (10 mM) arrested germination as observed in the wildtyp	17419840
AT1G58360	aap1-2	no obvious phenotype, endurance to toxic concentration of phenylalanine (2 mM) which did not influence development of roots and leaves of the mutant nor did a high level of phenylalanine (10 mM) arrested germination as observed in the wildtyp	17419840
AT1G58440	dry2	Plants are hypersensitive to drought. Mutants have decreased stomatal conductance when grown in low relative humidity. Root architecture is affected: roots are 60% shorter than wild type and produce more lateral roots. Lateral roots are shortr and branched.Plants are shorter and leaves are yellow green. Chemically, the mutants have decreased sterols in the roots, decreased chlorophyll levels and increased levels of proline.	19309460
AT1G58440	

AT1G69640	sbh1-1 sbh2-1	Dwarf plants with reduced vigor, dies before expansion of the true leaves, necrotic lesions on cotyledons. When grew on agar plates, the double mutants displayed extended viability and developed into compact and epinastic rosette with small curled leaves and short petioles, however no bolting. Short roots on agar plate.	18612100
AT1G70070	ise2-1	Homozygous mutant seeds can germinate in vitro on agar plates containing Murashige and Skoog (MS) salt plus 1% sucrose, although the  growth/developmental rate of the homozygous seedlings was slower compared with that of wild-type seedlings. Whereas wild-type seedlings displayed fully expanded green cotyledons, true leaves and a well-developed root system by 14 d after germination, homozygous seedlings were much smaller and white or pale green in color. Underdeveloped cotyledons emerged from the homozygous seeds but expanded very slowly and sometimes showed bumpy irregular surfaces and occasional anthocyanin expression. 

AT1G78240	tsd2-1	Increased activity of axial meristems, reduced root growth and enhanced de-etiolation.	17461780
AT1G79580	SALK_143526C	Additional columella root cap and lateral root cap cell layers in the mature embryo and postembryonic root development. Root meristem length and cell number is similar to WT. Extra layer of small, stem-cell-like cells, below the columella and lateral root cap stem cells.	19081078
AT1G79580	SALK_143526C	Delayed maturation and shedding of lateral root cap cells. The root cap extends into the differentiation zone.	20197506
AT1G79580	smb-3; brn1-1; brn2-1	Root cap cells appear as a mass lacking uniform morphology. Cells can be highly elongated and twisted.   Cells fail to stop elongation and to  adopt the rigid, rectangular morphology of COL cells.   When compared with smb-3, the triple mutant has thesame number of complete LRC layers, and the appearance of the cells in those layers closely resembles that in smb-3. However, partial remains of older LRC lay

AT2G01830	wol-3	Adventitious roots are normal.	16357038
AT2G01830	wol-3	Does not produce lateral roots from the primary root. However, 10 days after germination, lateral roots are formed at the upper part of the hypocotyl rather than at the root-hypocotyl junction.	16357038
AT2G01830	wol-3	Normal vascular system in adventitious roots, although the hypocotyl and primary root develop aborted vascular tissues.	16357038
AT2G01830	wol-3	Reduction in primary root length (about 5-fold compared to wildtype).	16357038
AT2G01940	sgr5-1	Inflorescence stems of mutant plants showed abnormal gravitropic response, while gravitropism in hypocotyls and roots was not altered.	9210330
AT2G01940	sgr5-3	Inflorescence stems of mutant plants showed abnormal gravitropic response, while gravitropism in hypocotyls and roots was not altered.	16813575
AT2G02220	pskr1-1	When grown on B5 agar, the mutant seedlings were indistinguishable from wildtype, albeit with a slight reduction in root growth. After three weeks

AT2G21790	cls8-1	Homozygous cls8-1 plants had crinkled leaves that showed slight yellowing at 4C  (also apparent under normal growth conditions)  Germination and cotyledon development are indistinguishable from wild type (Columbia), however the first developing true leaves appear bleached. Subsequent leaves emerge curled and ?ruffled?, with bleached edges. The rosette leaves expand some become crinkled and show patches of white or pits on the surface. There is often a degree of asymmetry along the axis of the main vein, with one half of the leaf failing to fully expand. The bleaching is localized to the petiole and the base of the leaf, and extends along the edges of the leaf. As they continue to expand, the leaves become greener, although the crinkling remains. Following bolting, the  cauline leaves show similar characteristics to rosette leaves.  Flowers produced by cls8-1 plants can be asymmetrical and the petals crinkled in appearance. Siliques grow  to a normal size, but are sligh

AT2G29130	SALK_025690C	reduced root elongation under PEG-induced dehydration	16804053
AT2G30250	SAIL_529B11	WT phenotype.  Root bending and root elongation assays under salt, osmotic and ABA treatments showed no significant difference compared to WT.
AT2G30250	wrky25-1; wrky33-1	Double mutant roots were significantly shorter than WT toots when under salt-stress.	18839316
AT2G30330	blos1 rna1	Longer primary roots and increased number of lateral roots as compared to WT. Increased levels of PIN1 and PIN2 proteins on the plasma membrane.	20971704
AT2G30410	kis-T1	No cell division defects are found in the root.	12215519
AT2G30410	kis-T1	Root cortical cells and epidermal hypocotyl cells of the mutant are indistinguishable from the wild type under normal growth conditions.
AT2G32410	AXL/axl-1 axr1-12	Homozygous axr1-12 plants that are heterozygous for axl-1 display severe growth defects. These seedlings developed short, slow growing roots compared with wild-type roots, and were arrested at th

AT2G39800	p5cs1-4	p5cs1-4 mutants accumulate less proline in response to salt stress than wild-type seedlings. Their roots are hypersensitive to salt stress and they have increased evidence of oxdative stress and lipid peroxidation under salt stress than wild type plants. Some genes involved in ROS detoxification are differentially expressed in  WT and p5cs1-4 mutants.	17971042
AT2G39940	coi1-1	Seedlings have green leaves and well-developed  roots when grown on MS medium [mineral salts and sucrose, 3% (w/v), solidified with agar, 0.6% (w/v)] supplemented with 50uM methyl jasmonate.	9582125
AT2G39940	coi1-15	Seedlings have green leaves and well-developed  roots when grown on MS medium [mineral salts and sucrose, 3% (w/v), solidified with agar, 0.6% (w/v)] supplemented with 50uM methyl jasmonate.	9582125
AT2G39940	coi1-18	Seedlings have green leaves and well-developed  roots when grown on MS medium [mineral salts and sucrose, 3% (w/v), solidified with agar, 0.6% (w/v)] supplemented with 

AT2G46240	SALK_009534	Mutant plants exhibited earlier flowering and shorter vegetative and reproductive phases, producing more branched roots and inflorescences compared to wild-type plants. Senescence occurred earlier in mutants than the wild-type plants.	16636050
AT2G46240	SALK_058290	Mutant plants exhibited earlier flowering and shorter vegetative and reproductive phases, producing more branched roots and inflorescences compared to wild-type plants. Senescence occurred earlier in mutants than the wild-type plants.	16636050
AT2G46370	CS8072	JA-Ile levels are seven-fold lower in the jar1-1 mutant compared to wild type plants. The levels of JA, JA-Val, and JA-Leu are similar in the wild-type and mutant backgrounds, whereas the level of JA-Phe and JA-ACC are higher in the mutant. Although this mutant is moderately insensitive to MeJA in root growth assays, it is sensitive to JA-Ile.	15258265
AT2G46370	CS8072	The jar1-1 mutant shows moderate insensitivity to MeJA in root growth  inhibiti

AT3G06810	ibr3-1	Roots moderately resistant to elongation inhibition by 2,4-DB (2,4-Dichlorophenoxybutyric acid) but remained sensitive to inhibition by 2,4-D (2,4-Dichlorophenoxyacetic acid). Fewer lateral roots are produced without induction which becomes more pronounced in response to IBA (Indole-3-butyric acid) treatment.	17277896
AT3G06810	ibr3-11	Roots moderately resistant to elongation inhibition by 2,4-DB (2,4-Dichlorophenoxybutyric acid) but remained sensitive to inhibition by 2,4-D (2,4-Dichlorophenoxyacetic acid). Fewer lateral roots are produced without induction which becomes more pronounced in response to IBA (Indole-3-butyric acid) treatment.	17277896
AT3G06810	ibr3-2	Roots moderately resistant to elongation inhibition by 2,4-DB (2,4-Dichlorophenoxybutyric acid) but remained sensitive to inhibition by 2,4-D (2,4-Dichlorophenoxyacetic acid). Fewer lateral roots are produced without induction which becomes more pronounced in response to IBA (Indole-3-butyric acid) treatmen

AT3G13300	vcs-3	Smaller than the wild type, and produced  narrow, asymmetric leaves when grown at  22�C. All five mutant alleles showed identical phenotypic responses to growth temperature. When grown at 16�C, vcs mutants resembled the wild-type controls except their leaves were pointed at the apex, and prolonged growth revealed defects in apical dominance. By contrast, growth at 29�C resulted in a strongly enhanced phenotype. These high temperature-grown mutants were chlorotic, leaves were narrow and epinastic, and prolonged growth resulted in a  callus-like growth at the apex. vcs roots  appeared similar to those of wild type,  except that at high temperature root hairs  occasionally appeared swollen. vcs also produced decreased numbers of lateral roots.	14660546
AT3G13300	vcs-4	Smaller than the wild type, and produced  narrow, asymmetric leaves when grown at  22�C. All five mutant alleles showed identical phenotypic responses to growth temperature. When grown at 16�C, vcs mutants re

AT3G20630	per1	per1 plants show normal root hair development under control conditions but displayed an inhibited root hair elongation phenotype upon Pi deficiency. The per1 mutant exhibits a pleiotropic phenotype under control conditions, resembling Pi-deficient plants in several aspects.	19969521
AT3G20780	rhl3-1	Complete absence of root hairs, abnormal patterning of leaf epidermis, and dwarfism.	9165126
AT3G20840	plt1-1	The mutation causes extra cells in the columella root cap and the QC.	15454085
AT3G20840	plt1-2	The mutation causes extra cells in the columella root cap and the QC.	15454085
AT3G20840	plt3-1/plt2-2/plt1-4	Rootless.Embryonic root differentiates 3 days after germination. Adventitious roots arrest 6 days after germination.	17960244
AT3G22200	CS6385	Hypersensitivity to NaCl and LiCl but not to mannitol in root growth assays. Under NaCl stress, pop2-1 mutant did not overaccumulate Na+ and Cl- but showed a slight decrease in K+ content. Additionally, mutant roots exhibited

AT3G28860	mdr1-2	Mutant mdr1-2 seedlings show an 80% reduction in acropetal auxin transport, but have normal levels of basipetal transport in their primary roots. Although gravitropism occurs normally in these mutants, they have a pronounced increase in root waviness resulting in a threefold increase in absolute curvature  relative to wild type seedlings.
AT3G30775	pdh1-2	Decreased root elongation and decreased dry weight for plants grown under low water potential conditions.	21791601
AT3G42170	daysleeper	Seedlings homozygous for the T-DNA insertion completely lacked expression of DAYSLEEPER. Mutant seedlings grew very slowly, and showed no expansion of the cotyledons or development of normal leaves or floral organs. The root remained short and had an excess of abnormal root hairs. The main and lateral root meristems appeared to have a normal organization when young, but as they matured became increasingly disorganized. After prolonged growth on plates, a few seedlings (1%) produced so

AT3G54140	atptr1-2 atptr5-1	The ptr1-2 ptr5-2 double mutant grows normally on soil and on AM medium. Its pollen tubes germinate and elongate normally on germination media. The tubes can germinate normally in the presence of 100uM alanyl-ethionine (Ala-Eth), a toxic dipeptide, but they elongate more than wild type pollen tubes.  Root growth of ptr1-2 ptr5-1 mutants exceeds wild type root growth when Ala-Eth is present in the growth media. The dry weight of ptr1-2 ptr5-1 double mutants is lower than that of wild type plants when both are grown with Pro-Ala or Ala-Ala dipeptides as the nitrogen source.	18753286
AT3G55830	SALK_014479	Mutant plants have dramatically reduced growth habit, defects in vascular formation and reduced cell-cell adhesion properties in hypocotyl and cotyledon tissues. Secondary growth is enhanced in epc1 hypocotyl tissues. When germinated on MS agar plates, cotyledon and leaf expansion, primary root elongation and hypocotyl elongation of the mutant seedlings were r

AT4G02280	sus2 sus3	Activity of SUS was reduced by about one-third in roots of the double mutant relative to Col-0, but otherwise the mutant was indistinguishable from Col-0.	17257168
AT4G02780	ga1-3 (Col-0)	GA-signaling promoting root and leaf expansion is not completely abolished.	17194763
AT4G05530	ibr1-1	The ibr1 mutants are resistant to the inhibitory effects of indole-3-butyric acid on root elongation, but their roots show the normal response to indole-3-acetic acid.	18725356
AT4G05530	ibr1-2	The ibr1-2 mutants are resistant to the inhibitory effects of intermediate levels of indole-3-butyric acid on root elongation, but their roots show the normal response to indole-3-acetic acid, NAA, and 2.4-D. Their root growth is also resistant to 2,4-DB, a compound that can be processed to produce 2,4-D via beta-oxidation. Lateral root proliferation in response to IBA is also reduced dramatically in ibr1-2 mutants, although they still produce normal numbers of lateral roots in response to N

AT4G24670	wei8-1 tar2-2	ACC-triggered dark-grown apical hook formation is compromised in this mutant. Root meristematic cells differentiate in this mutant, leading to a loss of the stem cell niche and the cessation of root growth. This mutant also lacks valves in the gynoecia.	18394997
AT4G24730	atnrt3.2-1	When grown on plates containing 250 &#956;M NO3- as the sole source of available N for 10 days, mutant plants grew poorly, shoot growth being particularly affected. Roots and shoots accumulated significantly less NO3- than wild type. Nitrate influx into roots of mutant plants was reduced.	16415212
AT4G26080	CS22	Mutant showed a similar sensitivity to 25 mM trehalose as wild type (inhibition of root growth).	17031512
AT4G26080	abi1td	mutant showed ABA-hypersensitive regulation of seed germination and root growth; increased sensitivity to salt and osmotic stress during germination; increased sensitivity to glucose during germination; reduced leaf transpiration rate; increased ABA-induc

AT4G32040	SALK_000339	No visible root phenotype.	16463096
AT4G32700	SALK_018851	Similar to teb-1, with a typical short-root phenotype and the aerial system that is reduced in size, with highly serrated and asymmetric leaves and a fasciated stem.	16517762
AT4G32700	SALK_035610	Similar to teb-1, with a typical short-root phenotype and the aerial system that is reduced in size, with highly serrated and asymmetric leaves and a fasciated stem.	16517762
AT4G32700	teb-1	A short-root phenotype with the phenotype in the aerial part that is similar to those of the tsk mutant. The roots in mutant seedlings were approximately one-third as long as those from wild type plants. This reduced growth became more pronounced at later time points. The root apical meristem was disorganized, and individual cell types, such as QC or initial cells, were difficult to discern. The aerial part of teb-1 plants also showed defects in morphology, such as abnormal phyllotaxy, highly serrated and asymmetric leaves and

AT5G01410	pdx1.3	When grown on media devoid of vitamin B6, the mutant seedlings have a reduced shoot growth and a distinct pale green phenotype, and a marked reduction in primary root growth.	17227548
AT5G01410	rsr4-1	Early development of lateral roots, often at the base of the hypocotyl. In the absence of vitamin B6, it was also observed that lateral roots frequently initiated closer to each other in comparison with the wild type.
AT5G01410	rsr4-1	Mutant plants display several prominent aerial and root phenotypes. Flowering time under long-day conditions (16 h light) was  delayed, in comparison with wild-type C24 plants, but mutant plants displayed no change in leaf number at the time of flowering, with an average of 22 rosette leaves. However, leaf size was reduced in the mutant, resulting in  decreased rosette growth and reduced fresh weight.	16766694
AT5G01410	rsr4-1	Strong reduction in primary root growth in sterile culture on minimal medium. Addition of different vitamins to the 

AT5G03860	mls-1	Mutants have only a slightly stunted phenotype when grown in the absence of exogenous sugar in the light. The phenotype is more obvious in the dark where hypocotyl elongation is inhibited, and little root development takes place.	15272001
AT5G03860	mls-2	Mutants have only a slightly stunted phenotype when grown in the absence of exogenous sugar in the light. The phenotype is more obvious in the dark where hypocotyl elongation is inhibited, and little root development takes place.	15272001
AT5G08020	SALK_088429C	This mutant does not differ visibly from a wild type plant when grown under normal conditions. A UV-B dose during the first week of growth on normal media affects the atrpa70b mutant and wild-type seedlings similarly, although the mutants have an increased level of leaf yellowing. After a longer period of exposure to UV-B, the level of chlorophyll A and B is reduced to a greater extent in rpa70b mutants than in wild-type, and root growth is more inhibited in the 

AT5G13570	dcp2-1	The homozygous progeny is seedling lethal, showed arrested postembryonic development including cotyledon expansion, development of vascular networks, root elongation, and shoot development.	17485080
AT5G13570	dcp2-1	The mutant produced no leaves, short roots  with short and swollen root hairs, chlorotic cotyledons that accumulated anthocyanins around their margins, and a short and swollen hypocotyl. In addition, the cotyledon-hypocotyl  junction was extremely fragile, which caused cotyledons to frequently fall off, even with gentle handling.	17513503
AT5G13570	tdt-1	The mutant produced no leaves, short roots  with short and swollen root hairs, chlorotic cotyledons that accumulated anthocyanins around their margins, and a short and swollen hypocotyl. In addition, the cotyledon-hypocotyl  junction was extremely fragile, which caused cotyledons to frequently fall off, even with gentle handling.	17485080
AT5G14420	rglg1 rglg2	Auxin (IAA) concentration was almost twofold lo

AT5G20730	CS24625	The phenotype of the double mutant is most obvious at its seedling stage, with its most prominent phenotype being severely impaired  lateral root formation. Its primary roots fail to produce lateral roots in 2-week-old seedlings. However, double mutant seedlings start to generate several lateral roots after ~2 weeks of growth, and their morphological appearance is normal.	15659631
AT5G20730	CS24625	nph4-1 arf19 double mutant;  agravitropic response in both hypocotyls and roots; impaired phototropic response in hypocotyls; impaired lateral root formation; small plant size; small and epinastic rosette leaves; reduced auxin sensitivity	15659631
AT5G20730	CS24626	The double mutant mutant displays agravitropic responses in both hypocotyls and roots. When seedlings are grown vertically under dark conditions, the regulation of growth orientation is disrupted in both hypocotyls and roots, with the hypocotyls occasionally growing downward and the roots upward. Also, the roots 

AT5G23720	phs1-1	phs1-1 exhibits phenotypes indicative of compromised cortical microtubule functions, such as left-handed helical growth of seedling roots, defective anisotropic growth at low doses of microtubule-destabilizing drugs, enhancement of the temperature-sensitive microtubule organization1-1 phenotype, and less ordered and more fragmented cortical microtubule arrays compared with the wild type.	15208393
AT5G24270	sos3-1 siz1-1	Decrease of macronutrients concentration in the MS salt formulation to 1/20X results in a substantial reduction in primary root growth.	15894620
AT5G24270	sos3-1 siz1-1	Mutant grown in greenhouse or growth chamber exhibits reduced shoot and root biomass relative to the wild-type.	15894620
AT5G24270	sos3-1 siz1-1	Substantially more pronounced prototypical Pi starvation root architecture responses than sos3-1 or wild-type seedlings. On Pi-limited medium, mutant seedlings exhibit an inhibition of primary root growth, an increase of lateral root development

AT5G35750	stip-d ahk2-2 ahk3-3 cre1-12	Small SAM and limited root growth.	20110319
AT5G37830	SALK_078745C	oxp1-1 cannot convert 5-oxoproline (5OP) into glutamine. Elevated levels of 5OP accumulate in the the leaves, stems, roots, flowers, and siliques of this mutant, while the level of glutamate is reduced. oxp1-1 mutants have normal seedling root growth and do not display any other morphological defects. They appear to have normal levels of Asp, Ser, Gln, Gly, Thr, Arg, and Ala, and have a wild-type sensitivity to nitrogen deficiency.	18768907
AT5G39510	sgr4-1	Inflorescence stems and hypocotyls of mutant plants showed abnormal gravitropic response, while root gravitropism was normal.	9210330
AT5G39610	nac2-1	Not significantly affected in lateral root development.	16359384
AT5G42890	Atscp2-1	Lower germination frequency and time, restored by addition of 1% sucrose. Smaller cotyledon rosettes and  shorter hypocotyls than wild-type and the root elongation  was also retarded. In the dark, 

AT5G47990	SALK_143113C	thad1-1 mutants have longer roots and elevated levels of thalianol and thalian-diol in root extracts compared to wild type plants. However, these mutants do not seem to have higher susceptibility to Atlernaria brassicicola or Pseudomonas syringae pv. tomato DC3000 based on assays performed on seedling roots.	18356490
AT5G48000	SALK_148229C	thah1-1 mutants have an elevated accumulation of thalianol and longer roots than wild type plants. Thalian-diol and desaturated thalian-diol are lost from the root extracts of thah1-1 mutants. However, these mutants do not seem to have higher susceptibility to Atlernaria brassicicola or Pseudomonas syringae pv. tomato DC3000 based on assays performed on seedling roots.	18356490
AT5G48000	thah1-2	thah1-2 mutants have elevated levels of thalianol in their root extracts when compared to wild type plants.	18356490
AT5G48010	thas1-1	thas1-1 mutants do not produce detectable levels of thalianol in root extracts but low  levels of thi

AT5G54690	irx8-2	Collapsed xylem vessels in roots and stems, suggesting a decrease in xylem wall strength in the homozygous mutants.	17237350
AT5G54690	irx8-2	In the homozygous mutant cell wall, decrease of cellulose content (~25%) and notable reduction in xylose in stems and roots (>50%).	17237350
AT5G54690	irx8-2	Where in wild-type roots xylem files form a closely packed array of cell  files, leading to the regular occurrence of  tripartite cell corners between xylem cell files, the central cylinder of homozygous mutant roots was disorganized and  the xylem files did not form a closely packed array. As a  consequence, tripartite xylem cell  corners could not be detected in the mutant.	17237350
AT5G54690	irx8-3	Collapsed xylem vessels in roots and stems, suggesting a decrease in xylem wall strength in the homozygous mutants.	17237350
AT5G54690	irx8-3	Where in wild-type roots xylem files form a closely packed array of cell  files, leading to the regular occurrence of  tripartite cell c

AT5G60920	CS8541	Callus-induced roots from the mutant displayed the mutant phenotype when they grew under restrictive conditions. All the mutants responded to a gravitropic stimulus in a manner similar to wild type although the change of the direction of growth was slower than wild type.	7743935
AT5G60920	CS8541	When grown on media containing 0.5% sucrose, the mutant had reduced root length but no apparent radial cell expansion.	7743935
AT5G60920	cob-4	Dark grown hypocotyls have reduced length compared to wild type. Root cells appear swollen and have abnormal radial expansion. Epidermal cells bulge. Cotyledon and hypocotyls are smaller and thicker than wild type. Cellulose microfibrils are randomly oriented.	15849274
AT5G61070	SALK_006938	Increased root hair density.	16176989
AT5G62165	agl42-4	No root anatomy phenotype	15937229
AT5G62500	eb1b-1	Plants grown on agar plates have roots that skew towards the left occasionally forming clockwise loops with twisted epidermal cell files. Also 

DER8	der8	Root hairs display very short stumps or collapsed root hairs.  Root hairs are often wavy, with an irregular hair diameter.  Sometimes develop bulbous structures at the root hair base.	15863435
ENL7	der3	Root hairs form in a regular pattern.  Length of the hair proper is not affected but side branches develop from the enlarged bases.	15863435
LIT	lit-1	Callus-induced roots from the mutant displayed the mutant phenotype when they grew under restrictive conditions. All the mutants responded to a gravitropic stimulus in a manner similar to wild type although the change of the direction of growth was slower than wild type.	7743935
LPI1	lpi1	Insensitive to the primary root growth inhibition caused by low phosphorous.The aerial parts are phenotypically similar to the wild type, except that these plants flowered 3-5 days earlier than wild type plants. Have normal root elongation and cell division.	16443695
LPI2	lpi2	Insensitive to the primary root growth inhibition caused by low phos

Initially I used the code shown above:

"grep" is used to look for a certain set of characters in a file.

"-E" allows the use of extended regular expressions

"-ia" means insensitive to case and treat any binary file as a text file. 

'roots?,?\.?\s' is the set of characters we are looking for. It stands for "root"  followed by an optional "s" (that way I also find the word "roots"), a dot or a coma optionally and a space. I could have just used 'root' as a query, but it would also find other derivatives of the word "root", which may be undesired.


However, "root" can appear as part of the germplasm name, although I think it is unlikely. As we only want to find "root" inside the phenotype description, I used the following code to make sure that "root" is only searched for in the phenotype description:


In [11]:
grep -P -ia '^.+\t.+\t.+?roots?,?\.?\s' Locus_Germplasm_Phenotype_20130122.txt

AT1G01550	bps1-2	Growth of bps1-2 mutants on CPTA-supplemented medium resulted in partial rescue of both leaf and root defects. bps1-2 mutants grown on control growth medium have small radialized leaves with very little vascular tissue, and very short misshapen knotted-looking roots. By contrast, bps1-2 mutants grown on CPTA-supplemented medium produced larger flattened leaves that contained primary and secondary veins, and smooth elongated roots.	17217459
AT1G01550	bps1-2	Mutant roots were abnormal; primary and lateral roots were short, root hairs formed close to the root apex, and root defects were most severe when plants were grown at low temperature.	15458645
AT1G01550	bps1-2	Under high light conditions (approximately 200 &mu;E.m-2.sec-1), CPTA-treated seedlings were completely photobleached and mutants showed rescue of both leaf and root development. Under moderate (approximately 100 &mu;E.m-2.sec-1) and low (30 &mu;E.m-2.sec-1) light intensities, seedlings are incompletely photob

AT1G08430	Atalm1-KO	On Al3+ medium, the mutant has significantly shorter roots than the Col wild type. In the absence of aluminum stress, the roots of mutant and wild type grew similarly.	16740662
AT1G08430	Atalm1-KO	The mutant exhibits very low levels of root  growth and malate release in the presence of  Al (0.2 nmol plant-1 24 h-1).
AT1G09100	rpt5b	Insensitive to high-glucose repression of cotyledon expansion, chlorophyll accumulation, true-leaf development and root elongation. The response is specific to glucose but not to osmotic changes.	17081979
AT1G09110	rpt5b	Insensitive to high-glucose repression of cotyledon expansion, chlorophyll accumulation, true-leaf development and root elongation. The response is specific to glucose but not to osmotic changes.	17081979
AT1G10130	eca3-1	Shoot and root growth is strongly retarded when plants are grown in the absence of Mn.  Also shows delayed growth when grown in the absence of Ca.	18024560
AT1G10570	ots1-1 ots2-1	The ots1-1 ots2-1 doubl

AT1G19220	CS24626	The double mutant mutant displays agravitropic responses in both hypocotyls and roots. When seedlings are grown vertically under dark conditions, the regulation of growth orientation is disrupted in both hypocotyls and roots, with the hypocotyls occasionally growing downward and the roots upward. Also, the roots and hypocotyls of the double mutant show reduced gravitropic curvatures compared with the wild type when vertically dark-grown seedlings are reoriented by 90�.	15659631
AT1G19220	CS24626	The phenotype of the double mutant is most obvious at its seedling stage, with its most prominent phenotype being severely impaired  lateral root formation. Its primary roots fail to produce lateral roots in 2-week-old seedlings. However, double mutant seedlings start to generate several lateral roots after ~2 weeks of growth, and their morphological appearance is normal.	15659631
AT1G19220	CS24626	nph4-1 arf19 double mutant;  agravitropic response in both hypocotyls and roots

AT1G22920	csn5a-1	Shorter hypocotyls in the dark, shorter  roots, smaller flowers, a decreased number of root hairs in response to jasmonic acid treatment, and altered light and auxin  responses.	17307927
AT1G22920	csn5a-2	Shorter hypocotyls in the dark, shorter  roots, smaller flowers, a decreased number of root hairs in response to jasmonic acid treatment, and altered light and auxin  responses.	17307927
AT1G23010	lpr1-1	Long primary root when grown on low Pi.	17496893
AT1G23010	lpr1-2	Long primary root when grown on low Pi.	17496893
AT1G23310	ggt1-1 (LUC)	In the absence of ABA treatment, root elongation of ggt1-1 was reduced nearly 50% compared to wild-type.	17318317
AT1G23320	tar1-1	This mutant does not have any obvious morphological defects and it responds normally to ACC and IAA in hypocotyl and root elongation assays.	18394997
AT1G23320	wei8-1 tar1-1 tar2-1	These triple mutants do not make a primary root, they have an extremely reduced hypocotyl, and they lack discernible vascul

AT1G31880	brx-2	lateral root formation in the mutant is insensitive to exogenous cytokinin; altered auxin response in lateral root primordium in the presence of exogenous cytokinin.	19037657
AT1G31880	brx-3	enhanced response to ABA-mediated inhibition of root growth	19201913
AT1G31880	brxS	The roots of brxS seedlings are as short, both when grown in the light or in darkness. In contrast to the root system, the shoot system morphology and flowering time of brxS plants resembles the Sav-0 shoot system, which was used as the control. Cell elongation and cell production rate in the root meristematic and elongation zone were decreased in brxS seedlings, contributing approximately one-third and two-thirds, respectively, to the overall difference in root length as compared with Sav-0 seedlings.	15031265
AT1G32450	SALK_005099 HOMOZYGOUS	Defective in xylem transport of nitrate from root to shoot. Morphologically WT when grown on soil. When grown on plates, roots were shorter than WT	18780802
AT

AT1G53700	wag1-2/wag2-1	Morphologically similar to double mutant wag1-1/wag2-1. When seedlings were grown on vertical plates, root exhibited pronounced wavy pattern. Gravitropism was not affected in mutant plants. Double mutant plants were more resistant to inhibition of root curling by auxin transport inhibitor NPA than wild type plants.	16460509
AT1G53940	SALK_025414C	has increased number of lateral roots, hypocotyls have impaired gravitropic curvature	19146828
AT1G53940	SALK_109449C	has increased number of lateral roots, hypocotyls have impaired gravitropic curvature	19146828
AT1G54180	SALK_017909	No visible root phenotype.	16514016
AT1G54960	anp2 anp3	Delayed growth,short roots, multiply branched root hairs.Radially expanded root and hypocotyl cells.Aberrant microtuble organization and resistance to oryzalin.	20215588
AT1G54990	CS8018	auxin resistant; irregular rosette leaves, slightly curled around leaf axis; roots elongate on auxin-containing medium; defective root gravitropism; 

AT1G64440	CS2257	Decrease root length (54% of that of wildtype) when grown on D-galactose-free medium.	12419184
AT1G64440	reb1-1	Bulging epidermal cells on the roots. The percentage of these cells appears to be dependent on environmental conditions (light intensity for example). This phenotype is not observed in the meristem but is first observed basal of the meristem.	
AT1G64440	reb1-1	Decreased root length (67% of that of wildtype) when grown on D-galactose-free medium.	12419184
AT1G64440	reb1-1	Less galactose in root cell wall compared to wildtype. No difference in shoot cell wall monosaccharide composition.	16500990
AT1G64440	reb1-1	Structural changes of xyloglucan in roots, not in shoots.	16500990
AT1G64440	reb1-2	At the restrictive temperature, epidermal bulging is observed and roots fail to grow straight in the elongation zone although this mutation does not appear to abolish gravitropic sensitivity.	
AT1G64520	rpn12a-1 (C24)	The rpn12a-1 mutant produces both the WT RPN12a prote

AT1G70560	wei8-1 tar2-2	ACC-triggered dark-grown apical hook formation is compromised in this mutant. Root meristematic cells differentiate in this mutant, leading to a loss of the stem cell niche and the cessation of root growth. This mutant also lacks valves in the gynoecia.	18394997
AT1G70560	wei8-2	The hypocotyl of this mutant responds normally to ACC when grown in the dark, but, its roots are moderately less sensitive to ACC than wild type seedlings grown under the same conditions. This difference in sensitivity can be eliminated when the mutants are treated with ACC and low levels of IAA. wei8-2 and wild type hypocotyls and roots respond similarly to IAA in the absence of ACC.	18394997
AT1G71100	rsw10	Bulging of epidermal cells is observed in roots but not in the hypocotyl.	17059404
AT1G73360	edt1	edt1 mutant plants show enhanced drought tolerance due to overexpression of the HDG11 protein. edt1 mutants have a longer primary root with more lateral roots, and have a higher root bi

AT2G01830	cre1-12	Reduced cytokinin-induced inhibition of adventitious root formation compared to wildtype.	
AT2G01830	cre1-12	Reduced sensitivity to cytokinin in root growth assay (exogenous application of cytokinins inhibits wildtype root elongation).	
AT2G01830	cre1-12 ahk2-2tk	Reduced sensitivity to cytokinin in root growth assay (exogenous application of cytokinins inhibits wildtype root elongation).	
AT2G01830	cre1-12 ahk2-2tk ahk3-3	Shoot and root growth is very slow and leaf number is decreased compared to wildtype.	
AT2G01830	cre1-12 ahk2-2tk ahk3-3	Shorter root than wildtype plant.	
AT2G01830	cre1-12 ahk2-2tk ahk3-3	The size and activity of the root apical meristem is decreased with respect to wildtype.	
AT2G01830	cre1-12 ahk3-3	No cytokinin-induced inhibition of adventitious root formation.	
AT2G01830	cre1-12 ahk3-3	Reduced sensitivity to cytokinin in root growth assay (exogenous application of cytokinins inhibits wildtype root elongation).	
AT2G01830	cre1-2	Resistant to cyt

AT2G20000	hbt_2311	The roots in mutants was more than ten-fold shorter in length than the roots in the wild-type (Col-0).	9425146
AT2G20000	hbt_5721	The roots in mutants was more than ten-fold shorter in length than the roots in the wild-type (Col-0).	9425146
AT2G20000	hbt_5859	The roots in mutants was more than ten-fold shorter in length than the roots in the wild-type (Col-0).	9425146
AT2G20000	hbt_e65	Mutant plants have residual root meristem activity (weak allele). The root length in mutants was less than half of the root length in the wild-type (Col-0).	9425146
AT2G20190	clasp-1	Significant dwarfing at all stages of development compared with wild type.  Cell expansion in leaves, roots and hypocotyls is affected.	17873093
AT2G20890	thf1-1	thf1-1 Root Growth Is Hypersensitive  to Increased D-Glucose	16582010
AT2G21170	pdtpi	This mutant is severely stunted in growth. The seeds germinate at a rate comparable to wild type seeds but they begin to show aberrant phenotypes 3 to 4 days pos

AT2G27050	ebf1-3 ebf2-3 eil1-3	Modest rescue of the ebf1-3 ebf2-3 phenotype. Seedlings continue grow and develop at a slow pace after germination. Cotyledons partially expanded and turned pale green. Stunted root densely covered with root hair. Plants never flowered or set seeds.	17307926
AT2G27228	SALK_079402 homozygous	Reduced root vascular population.  Roots lose bilateral symmetry and are monarch instead of diarch.	17626058
AT2G27230	SALK_079402 homozygous	Reduced root vascular population.  Roots lose bilateral symmetry and are monarch instead of diarch.	17626058
AT2G27230	lhw-1	Reduced root vascular population.  Roots lose bilateral symmetry and are monarch instead of diarch.	17626058
AT2G27230	lhw-2	Reduced root vascular population.  Roots lose bilateral symmetry and are monarch instead of diarch.	17626058
AT2G27230	lhw-3	Reduced root vascular population.  Roots lose bilateral symmetry and are monarch instead of diarch.	17626058
AT2G27230	lhw-4	Reduced root vascular population.  

AT2G39940	coi1-1	Seedlings have green leaves and well-developed  roots when grown on MS medium [mineral salts and sucrose, 3% (w/v), solidified with agar, 0.6% (w/v)] supplemented with 50uM methyl jasmonate.	9582125
AT2G39940	coi1-15	Seedlings have green leaves and well-developed  roots when grown on MS medium [mineral salts and sucrose, 3% (w/v), solidified with agar, 0.6% (w/v)] supplemented with 50uM methyl jasmonate.	9582125
AT2G39940	coi1-18	Seedlings have green leaves and well-developed  roots when grown on MS medium [mineral salts and sucrose, 3% (w/v), solidified with agar, 0.6% (w/v)] supplemented with 50uM methyl jasmonate.	9582125
AT2G40220	CS3837	Mutant root growth was already somewhat reduced compared to WT plants under normal growth conditions but was significantly less sensitive to inhibition by 25 mM trehalose. Root length was reduced only by 46% compared to 90% in WT plants (Fig. 1B). The  germination rates on trehalose were identical to those on medium without trehalo

AT2G46370	CS8072	The jar1-1 mutant shows moderate insensitivity to MeJA in root growth  inhibition assays, but it has normal male fertility.
AT2G46370	jar1-11	The mutant has reduced sensitivity to methyl jasmonate in root growth inhibition experiements. It also produces lower levels of jasmonyl-isoleucine (JA-Ile) than wild-type plants in response to wounding.	18247047
AT2G46370	jar1-8	The jar1-8 mutant shows moderate insensitivity to MeJA in root growth  inhibition assays.
AT2G47000	mdr4-1	Although mutant seedlings display normal acropetal auxin transport in the root, basipetal auxin transport is reduced by about 50%. Root waving proceeds normally in the mutant seedlings, but gravitropic curvature is enhanced, occurring more quickly and to a greater degree than in wild type seedlings.	17557805
AT2G47000	mdr4-1	mdr4-1 mutants display normal root architecture and have wild type rates of lateral root elongation.	17557807
AT2G47000	mdr4-2	Although mutant seedlings display normal acropetal

AT3G08040	CS6585	defective in iron translocation; chlorotic plants; expression of iron deficiency responses under conditions of iron sufficiency; overaccumulation of iron and other metals (plants accumulate approximately two-fold excess iron, four-fold excess manganesse and two-fold excess zinc in their shoots); detached roots are capable of repressing iron uptake responses when cultured under iron-sufficient conditions; when shoots are appropriately supplied with iron, they regreen and the plants are capable of correctly regulating root iron uptake responses; isolated protoplasts have lower iron levels than those from wild type plants; plants accumulate abnormal levels of ferric iron in their root vasculature, FRD3 is expressed in the pericycle and other vascular cylinder cells in the mature portion of the root; lacks trichomes on stems and leaves.	12172022
AT3G08710	trx h9	Roots and leaves of  the homozygous mutants were significantly shorter and smaller  than wild-type. Root tips of

AT3G18780	act2-1	Mutants have abnormal root hairs that are stunted, bulging or may be branched. The orientation of microtubules is abnormal.	19304937
AT3G18780	act2-1/act7-4	Plants are dwarf and reach only 15% of the height of wild type. Roots lack root hairs- the initiate but do not develop further than a bulge. Leaves are small with reduced surface area and fewer lobes. Leaf trichomes are abnormal and contain fewer branches. Also defects in flower morphology, reduced fertility, abnormal inflorescence structure and abnormal silique development.	19304937
AT3G18780	act2-1/act8-2	Lacks root hairs.Root hairs initiate but do not develop further.	19304937
AT3G19050	pok1-1/pok2-1	At seedling stage pok1/pok2 double mutants differ from wild-type by having smaller cotyledons as well as shorter, wider roots and hypocotyls. Adult plants exhibit a dwarfed stature, but all organs are present, although miniaturized. Plants are fertile but produce reduced numbers of seeds. Microscopic observation rev

AT3G27920	sos3-1 siz1-1	Decrease of macronutrients concentration in the MS salt formulation to 1/20X results in a substantial reduction in primary root growth.	15894620
AT3G27920	sos3-1 siz1-1	Mutant grown in greenhouse or growth chamber exhibits reduced shoot and root biomass relative to the wild-type.	15894620
AT3G27920	sos3-1 siz1-1	Substantially more pronounced prototypical Pi starvation root architecture responses than sos3-1 or wild-type seedlings. On Pi-limited medium, mutant seedlings exhibit an inhibition of primary root growth, an increase of lateral root development and length, and higher root/shoot fresh weight ratio than wild-type.  Enhanced resistance to Pseudomonas syringae DC3000.  Smaller in stature.  Impaired drought tolerance.	15894620
AT3G28860	mdr1-1	Lateral roots initiate and emerge at normal rates in mdr1-1 mutants, but they elongate ~50% more slowly than wild type lateral roots, and have a 60% reduction in acropetal auxin transport. Primary root elongation is no

AT3G53900	upp-1	dramatic growth retardation; pale green to albino; flimsy roots with less branching; chloroplastic disorder.	19563437
AT3G54140	CS859493	The ptr1-1 mutant grows normally on soil and on AM medium. Its pollen tubes germinate and elongate normally on germination media and in the presence of 100uM alanyl-ethionine (Ala-Eth), a toxic dipeptide. Root growth of ptr1-1 mutants exceeds wild type root growth when Ala-Eth is present in the growth media. The dry weight of ptr1-1 mutants is lower than that of wild type plants when both are grown with Pro-Ala or Ala-Ala dipeptides as the nitrogen source.	18753286
AT3G54140	SALK_130803C	The ptr1-2 mutant grows normally on soil and on AM medium. Its pollen tubes germinate and elongate normally on germination media. The tubes can germinate and elongate normally in the presence of 100uM alanyl-ethionine (Ala-Eth), a toxic dipeptide. Root growth of ptr1-2 mutants exceeds wild type root growth when Ala-Eth is present in the growth media. T

AT3G63110	atipt3 atipt 5 atipt 7	Relative to wild type, both the number of lateral roots (longer than 1cm) and the total length of the lateral roots are increased.	17062755
AT4G00710	SALK_096500C	shorter hypocotyl length when grown in the dark on regular medium or medium containing the BR biosynthetic inhibitor brassinazole (BRZ).  reduced responses to BL in hypocotyl elongation, root inhibition, and expression of the BZR1 target gene DWF4 and the BES1 target gene SAUR-Ac.	18653891
AT4G01100	CS333468	Homozygotes have shorter roots than WT. Respiration rates are significantly lower than in WT roots.	18923018
AT4G01370	mpk4	Dwarf, delayed growth, short roots. Defective root hairs that are branched and swollen.Ectopic root hairs.In epidermal cells, cortical microtubules are bundled and disorganized.	20215588
AT4G02280	sus2 sus3	Activity of SUS was reduced by about one-third in roots of the double mutant relative to Col-0, but otherwise the mutant was indistinguishable from Col-0.	17257168

AT4G24670	wei8-1 tar2-2	ACC-triggered dark-grown apical hook formation is compromised in this mutant. Root meristematic cells differentiate in this mutant, leading to a loss of the stem cell niche and the cessation of root growth. This mutant also lacks valves in the gynoecia.	18394997
AT4G24730	atnrt3.2-1	When grown on plates containing 250 &#956;M NO3- as the sole source of available N for 10 days, mutant plants grew poorly, shoot growth being particularly affected. Roots and shoots accumulated significantly less NO3- than wild type. Nitrate influx into roots of mutant plants was reduced.	16415212
AT4G26080	CS22	Mutant showed a similar sensitivity to 25 mM trehalose as wild type (inhibition of root growth).	17031512
AT4G26080	abi1td	mutant showed ABA-hypersensitive regulation of seed germination and root growth; increased sensitivity to salt and osmotic stress during germination; increased sensitivity to glucose during germination; reduced leaf transpiration rate; increased ABA-induc

AT4G32040	SALK_000339	No visible root phenotype.	16463096
AT4G32700	SALK_018851	Similar to teb-1, with a typical short-root phenotype and the aerial system that is reduced in size, with highly serrated and asymmetric leaves and a fasciated stem.	16517762
AT4G32700	SALK_035610	Similar to teb-1, with a typical short-root phenotype and the aerial system that is reduced in size, with highly serrated and asymmetric leaves and a fasciated stem.	16517762
AT4G32700	teb-1	A short-root phenotype with the phenotype in the aerial part that is similar to those of the tsk mutant. The roots in mutant seedlings were approximately one-third as long as those from wild type plants. This reduced growth became more pronounced at later time points. The root apical meristem was disorganized, and individual cell types, such as QC or initial cells, were difficult to discern. The aerial part of teb-1 plants also showed defects in morphology, such as abnormal phyllotaxy, highly serrated and asymmetric leaves and

AT5G01410	rsr4-1	Strong reduction in primary root growth in sterile culture on minimal medium. Addition of different vitamins to the medium revealed that this phenotype was fully complemented by the provision of pyridoxine (vitamin B6) to the mutant.
AT5G03150	SALK_054242C	jkd-4 mutants have slightly shorter roots than wild type seedlings, and about 15% of them show premature lateral root initiation. They have fewer root meristem cells than wild type seedlings at 5 and 12 days post-germination. But, there is a higher number of cells in the endodermal and cortical layers of the root in jkd-4 mutants than in WT seedlings. A loss of expression of SCR in the QC can be detected in jkd-4 mutants at the early heart stage and torpedo stages of embryonic development. A GFP:SHR fusion protein, normally confined to the nuclei of QC cells, is present in the cytoplasm most QC cells in jkd-4 mutants. However, GFP:SHR is observed in the nuclei of ground tissue cells in jkd-4 mutants.	17785527
AT5G031

AT5G08100	CS67896	Increased sensitivity to delay in root hair emergence in response to exogenous asparagine. Increased accumulation of asparagine in root when grown on medium containing asparagine as a single nitrogen source. Increased seed size under conditions of nitrogen deficiency. Increased concentration of free asparagine and other free amino acids in seed when grown under low illumination.	21800258
AT5G08100	CS67896	increased sensitivity to inhibition of root elongation by exogenous asparagine	21800258
AT5G08100	SAIL_690_E07	increased sensitivity to inhibition of root elongation by exogenous asparagine	21800258
AT5G08100	SALK_087377C	increased sensitivity to inhibition of root elongation by exogenous asparagine	21800258
AT5G08550	ilp1-1	Shorter hypocotyls and roots compared to their wild type siblings in darkness.  In the light, plants have shorter hypocotyls and smaller cotyledons compared with the wild type, and there was inhibition of root elongation.  There is also a reducti

AT5G18170	CS3975	deficient in glutamate dehydrogenase; conditional phenotype: impaired root growth and mild shoot chlorosis under intermediate/high inorganic nitrogen conditions, this is exagerated when plants are grown in high concentrations of inorganic nitrogen under suboptimal conditions( e.g. without vitamins); seedling growth is specifically retarded on media containing exogenously supplied inorganic nitrogen.	8643469
AT5G18560	puchi-1	shorter visible lateral roots at 9 days after germination. The proximal region of each lateral root in the mutant is significantly swollen and often bent. Growth and morphology of the primary root of the puchi mutant is normal.	17630277
AT5G19040	atipt1 atipt 3 atipt 5 atipt7	External application of <i>trans</i>-zeatin partially rescues the growth of aerial parts of the mutant, and reduced its lateral root elongation.	17062755
AT5G19040	atipt1 atipt 3 atipt 5 atipt7	Reduced cambial activity and reduced secondary growth in both shoots and roots.	190

AT5G20730	CS24627	The double mutant mutant displays agravitropic responses in both hypocotyls and roots. When seedlings are grown vertically under dark conditions, the regulation of growth orientation is disrupted in both hypocotyls and roots, with the hypocotyls occasionally growing downward and the roots upward. Also, the roots and hypocotyls of the double mutant show reduced gravitropic curvatures compared with the wild type when vertically dark-grown seedlings are reoriented by 90�.	15659631
AT5G20730	CS24627	The phenotype of the double mutant is most obvious at its seedling stage, with its most prominent phenotype being severely impaired  lateral root formation. Its primary roots fail to produce lateral roots in 2-week-old seedlings. However, double mutant seedlings start to generate several lateral roots after ~2 weeks of growth, and their morphological appearance is normal.	15659631
AT5G20730	CS24628	The double mutant mutant displays agravitropic responses in both hypocotyls and

AT5G25350	ebf1-1 ebf2-1	When germinated in the dark, the mutant display a ctr1-like phenotype (short hypocotyls, short roots, and exaggerated apical hooks).	14675533
AT5G25350	ebf1-1 ebf2-1	When grown in the light, the double mutants display slightly longer hypocotyls, curled unexpanded cotyledons, and short and thicker roots with ectopic root hairs, similar to the ctr1 mutant. Moreover, the average area of the leaf epidermal cells was reduced 5-fold compared to wild type.	14675533
AT5G25350	ebf1-2 ebf2-1	ctr1-like phenotype (short hypocotyls, short roots, and exaggerated apical hooks).	14675533
AT5G25350	ebf1-3 ebf2-3 eil1-3	Modest rescue of the ebf1-3 ebf2-3 phenotype. Seedlings continue grow and develop at a slow pace after germination. Cotyledons partially expanded and turned pale green. Stunted root densely covered with root hair. Plants never flowered or set seeds.	17307926
AT5G27320	gid1a-1 gid1b-1 gid1c-1	Dramatic reduction in rosette radius and root length compared to wild typ

AT5G44070	cad1-1	When grown in the presence of inhibitory levels of cadmium, growth is inhibited, leaves become progressively chlorotic and the roots develop a distinct brown pigment.	7770517
AT5G44070	cad1-3	When grown in the presence of inhibitory levels of cadmium, growth is inhibited, leaves become progressively chlorotic and the roots develop a distinct brown pigment.	7770517
AT5G44070	cad1-4	When grown in the presence of inhibitory levels of cadmium, growth is inhibited, leaves become progressively chlorotic and the roots develop a distinct brown pigment.	7770517
AT5G44070	cad1-5	When grown in the presence of inhibitory levels of cadmium, growth is inhibited, leaves become progressively chlorotic and the roots develop a distinct brown pigment.	7770517
AT5G44740	SALK_129731	Homozygotes are sensitive to UV-B treatment. UV-B treatment causes an inhibition of root growth.	18494853
AT5G44750	SALK_005721	Mutant plants were moderately sensitive to UV-B irradiation than the wild type whe

AT5G49970	SALK_149382C	Reduced root growth and foliar biomass. Total vitamin B6 levels were lower than in wild type, measured using yeast bioassay.	17873088
AT5G49970	pdx3-1	Reduced root growth and foliar biomass. Total vitamin B6 levels were lower than in wild type, measured using yeast bioassay.	17873088
AT5G50200	Atnrt3.1-1	When grown on plates containing 250 &#956;M NO3-  as the sole source of available N for 10 days, mutant plants grew poorly, shoot  growth being particularly affected. Roots and shoots accumulated significantly less NO3- than wild type. Nitrate influx into roots of  mutant plants was reduced.	16415212
AT5G50375	cpi1-1	sterile, dwarf phenotype with small, round, dark leaves, a short hypocotyl and root and defective root gravitropism. Roots presented defects in directional orientation along the gravity vector, normal meristem size but a shorter elongation zone, reduced epidermal  cell length, delayed gravitropic  curvature and slower growth  than the wild-type
AT5G5

AT5G56290	CS3949	resistant to the inhibition of root elongation caused by 5 microM indole-3-butyric acid (IBA); fails to induce normal number of lateral roots when treated with 5 microM IBA; exhibits shorter hypocotyl than wt when grown in the dark without sucrose, but has a normal hypocotyl when grown in the dark with 0.5% sucrose; no visible phenotype in aerial morphology; defective in seed storage lipid utilization.	11063705
AT5G57390	EMKox-2	22.9% (72 of 314) of seedlings produced lightgreen, somatic embryos  at the tip margin of  cotyledons. Eight percent (25 of 314) of seedlings were equipped with ectopic trichomes on the  cotyledons.  Elongation of roots and emergence of root  hairs in AtEMKox-2 were severely inhibited in comparison  to WT. Furthermore,  the roots of AtEMKox-2 showed aberrant structures  such as tumescence of cells and entwined root hairs. In spite of the abnormal shape of roots, the root  apical meristem of AtEMKox-2 was morphologically  identical to WT.	204053

AT5G66880	snrk2.2 snrk2.3	snrk2.2 snrk2.3 was found to be ABA-insensitive in seed germination and early seedling growth  snrk2.2 snrk2.3 seeds showed green cotyledons at 5 d after sowing (non of the wild-type seeds had germinated at this time)  snrk2.2 snrk2.3 had decreased seed dormancy  snrk2.2 snrk2.3 was insensitive to the stimulating effect of low ABA concentration (<1 uM)on root growth  snrk2.2 snrk2.3 had a greatly reduced level of a 42-kD kinase activity capable of phosphorylating peptides from ABF (for ABA Response Element Binding Factor) transcription factors
AT5G67270	eb1c-1	Plants grown on agar plants show leftward skewing of root growth occasionally forming clockwise spirals with twisted epidermal cell files.	18281505
CUD	cud-1	Callus-induced roots from the mutant displayed the mutant phenotype when they grew under restrictive conditions. All the mutants responded to a gravitropic stimulus in a manner similar to wild type although the change of the direction of growth was 

"-P" means that the regular expression will be interpreted as defined by Perl. This is needed to be able to use "\t" to look for a tab character.

'^.+\t.+\t.+?root,?\.?\s' Has the following meaning:

^ --> start at the begining of the line.

.+ --> anything several times (it will match the locus name).

\t --> tab character (between the locus name and the germplasm name).

.+ --> anything several times (it will match the germplasm name).

\t --> tab character (between the germplasm name and the phenotype).

.+? --> optionally, anything several times (the phenotype description before the word "root". It is optional because "root" can be the first word).

roots?,?\.?\s -->  "root"/"roots" optionally followed by dot or coma and a space.

## Problem 7

In [8]:
grep -P -ia '^.+\t.+\t.+?roots?,?\.?\s' Locus_Germplasm_Phenotype_20130122.txt | grep -E -ioa '^at[0-9]g[0-9]+' > Root-associated-Loci.txt
cat Root-associated-Loci.txt

AT1G01550
AT1G01550
AT1G01550
AT1G02970
AT1G03000
AT1G03060
AT1G03060
AT1G03060
AT1G03060
AT1G04020
AT1G04030
AT1G04160
AT1G04160
AT1G04160
AT1G04220
AT1G04250
AT1G05180
AT1G05180
AT1G05850
AT1G05850
AT1G06290
AT1G06290
AT1G08030
AT1G08260
AT1G08370
AT1G08430
AT1G09100
AT1G09110
AT1G10130
AT1G10570
AT1G11130
AT1G15550
AT1G15690
AT1G16150
AT1G17110
AT1G17170
AT1G17580
AT1G17580
AT1G18800
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G21700
AT1G21970
AT1G21970
AT1G22920
AT1G22920
AT1G23010
AT1G23010
AT1G23310
AT1G23320
AT1G23320
AT1G26870
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27760
AT1G28560
AT1G29990
AT1G31480
AT1G31880
AT1G31880
AT1G31880
AT1G31880
AT1G32450
AT1G33280
AT1G33280
AT1G33290
AT1G33290
AT1G34370
AT1G34370
AT1G34370
AT1G35580
AT1G47056
AT1G47720
AT1G47720
AT1G48380
AT1G48920
AT1G49240
AT1G49240
AT1G49240
AT1G51190
AT1G52340
AT1G53700
AT1G53700
AT1G53700


The code before the pipe (|) was explained in Problem 6.

The pipe allows to use the output of the first part of the code as the input of the second half.

The second half of the code uses a grep to search for the AGI Locus Code

"-E" allows the use of extended regular expressions.

"-ioa" means insensitive to case, print only the match and treat any binary file as a text file. 

'^at[0-9]g[0-9]+' describes the structure of the AGI Locus Code:

^ --> start at the begining of the line.

at --> characters "at" (they can also be in capital letters due to the aforementioned use of -i).

[0-9] --> any number.

g --> character g (or G).

[0-9]+ --> any number several times.


"> Root-associated-Loci.txt" sends the output to the file Root-associated-Loci.txt

## Problem 8

In [9]:
grep -P -ia '^.+\t.+\t.+?roots?,?\.?\s' Locus_Germplasm_Phenotype_20130122.txt | grep -E -ioa '[0-9]+$' > Root-associated-Publications.txt
cat Root-associated-Publications.txt

17217459
15458645
17217459
17209125
14745029
19392685
19392685
19392685
19392685
18591352
18591352
18178669
18178669
19060218
18786002
16367964
17655650
7704045
7743935
7743935
15743450
15743450
19666544
16278345
17485080
17081979
17081979
18024560
18849491
15618487
16460513
16210544
16286448
18485060
18178669
19060218
15659631
15659631
15659631
15659631
15659631
15659631
15659631
15659631
16461383
12244233
12244233
17307927
17307927
17496893
17496893
17318317
18394997
18394997
19081078
16357038
16357038
16357038
11479382
16361392
16361392
16361392
20110319
19210750
16098103
19004800
8819871
15031265
19037657
19201913
15031265
18780802
20197506
20197506
17535918
17535918
17535918
21441406
17435085
17189341
17189341
16339310
17369435
19304937
19304937
19304937
17960244
17031512
16460509
16460509
16460509
16514016
20215588
7704045
7704045
7704045
7704045
17369372
16359390
21441406
7867930
7867930
7867930
19309460
17355438
17355438
18849491
18408046
16463096
16367966
16367966
12419184
124

The code before the pipe was explained in Problem 6.

The second half of the code uses a grep to search for the PubMed ID.

"-E" allows the use of extended regular expressions.

"-ioa" means insensitive to case, print only the match and treat any binary file as a text file.

'[0-9]+$' looks for several numbers at the end of the line (the PubMed ID). 

"> Root-associated-Loci.txt" sends the output to the file Root-associated-Loci.txt.


## Problem 9

In [13]:
grep -E -ioa '^at1' Root-associated-Loci.txt | wc -l

151


In [14]:
grep -E -ioa '^at2' Root-associated-Loci.txt | wc -l

109


In [15]:
grep -E -ioa '^at3' Root-associated-Loci.txt | wc -l

92


In [16]:
grep -E -ioa '^at4' Root-associated-Loci.txt | wc -l

53


In [17]:
grep -E -ioa '^at5' Root-associated-Loci.txt | wc -l

161


I do a grep for the file Root-associated-Loci.txt (containing the AGI Locus Code). In this code, the third character is a number (the chromosome number). That way, '^at1' will find the entries corresponding to loci inside the chromosome 1, '^at2' finds loci inside the chromosome 2 and so on.

The output of this grep is used as the input of wc -l (counts the number of lines). Its output is the number of root-related genes found in the chromosome. If it is greater than 0, there are genes associated with roots in the chromosome.

The hypothesis that genes associated with roots should be found on all chromosomes is true for this dataset as all the numbers obtained are greater than 0.